From: Simon McVittie <smcv@collabora.com>
Date: Mon, 7 Sep 2020 14:16:59 +0100
Subject: open-uri: Assert that find_recommended_choices yields non-NULL

choices is never NULL, but might be an array containing only NULL.

Signed-off-by: Simon McVittie <smcv@collabora.com>
Applied-upstream: 1.7.3
---
 src/open-uri.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/open-uri.c b/src/open-uri.c
index f746e64..c5b8705 100644
--- a/src/open-uri.c
+++ b/src/open-uri.c
@@ -669,6 +669,8 @@ handle_open_in_thread_func (GTask *task,
 
   /* collect all the information */
   find_recommended_choices (scheme, content_type, &default_app, &choices, &n_choices);
+  /* it's never NULL, but might be empty (only contain the NULL terminator) */
+  g_assert (choices != NULL);
   if (default_app != NULL && !app_exists (default_app))
     g_clear_pointer (&default_app, g_free);
   use_default_app = should_use_default_app (scheme, content_type);
